### PR TITLE
native, checker: cast float_literal to f32 explicitly in code gen

### PR DESCRIPTION
**Pull request by lemoncmd**

### CODE REVIEW

**Comment by lemoncmd** [comment]
should `float_literal (infix op) f32` promoted to f64 in calculation? 
```
	// vlib/v/tests/sizeof_test.v#L14
	assert r > 5.6e-45 && r < 5.7e-45
	// where r is f32
```
```
	println(typeof(f32(1.0) + 1.0).name) // f32
```

**Comment by spytheman** [comment]
Yes, use `f64` for `float_literal`. The C backend also does it.

**Comment by spytheman** [comment]
I would have been best if the checker or the transformer replaced all `float_literal` with `f64`, since it is entirely a frontend concept.

**Comment by lemoncmd** [comment]
I have to rewrite code calculating float in native backend then. I'll do it.

**Comment by lemoncmd** [comment]
Current C backend does implicit type conversion, and the checker converts types depends on it.
Since we don't have type assertion or calculation in the native backend, there will be a bunch of bugs from this inconsistency.

File: vlib/v/gen/native/tests/float.vv [hunk 1]
```diff
@@ -43,11 +43,11 @@ fn float_cast_test() {
 
 // TODO return float literal
```
Comment (axetroy) [inline vlib/v/gen/native/tests/float.vv:44]: I think this should be removed now

**Pull request merged**